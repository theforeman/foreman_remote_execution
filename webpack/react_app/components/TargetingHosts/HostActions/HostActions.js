import React from 'react';
import PropTypes from 'prop-types';

const HostActions = (props) => {
  const { actions } = props;

  if (actions === undefined || actions.length === 0) {
    return ('');
  }

  if (actions.length === 1) {
    return (<a href={actions[0].path} className="btn btn-sm btn-default">{actions[0].name}</a>);
  }
  return (
      <div className="btn-group">
        <span className="btn btn-sm btn-default">
          <a href={actions[0].path}>
            {actions[0].name}
          </a>
        </span>
        <a className="btn btn-default btn-sm dropdown-toggle" data-toggle="dropdown" href="#">
          <span className="caret"></span>
        </a>
        <ul className="dropdown-menu pull-right">
          { actions.slice(1, actions.length).map(link =>
            <li key={link.path}>
              <a href={link.path}>{link.name}</a>
            </li>)}
        </ul>
      </div>
  );
};

HostActions.propTypes = {
  actions: PropTypes.array,
};

export default HostActions;
