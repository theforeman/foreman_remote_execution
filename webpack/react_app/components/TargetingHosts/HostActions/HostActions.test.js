import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import HostActions from './HostActions';

const fixtures = {
  'renders HostActions with no actions': {
    actions: undefined,
  },
  'renders HostActions with 0 actions': {
    actions: [],
  },
  'renders HostActions with one action': {
    actions: [{ path: '/1', name: 'action one' }],
  },
  'renders HostActions with more actions': {
    actions: [{ path: '/1', name: 'action one' },
      { path: '/2', name: 'action two' }],
  },
};
describe('HostActions', () =>
  testComponentSnapshotsWithFixtures(HostActions, fixtures));
