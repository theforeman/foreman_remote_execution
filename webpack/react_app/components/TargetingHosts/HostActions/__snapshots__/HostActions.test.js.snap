// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostActions renders HostActions with 0 actions 1`] = `""`;

exports[`HostActions renders HostActions with more actions 1`] = `
<div
  className="btn-group"
>
  <span
    className="btn btn-sm btn-default"
  >
    <a
      href="/1"
    >
      action one
    </a>
  </span>
  <a
    className="btn btn-default btn-sm dropdown-toggle"
    data-toggle="dropdown"
    href="#"
  >
    <span
      className="caret"
    />
  </a>
  <ul
    className="dropdown-menu pull-right"
  >
    <li
      key="/2"
    >
      <a
        href="/2"
      >
        action two
      </a>
    </li>
  </ul>
</div>
`;

exports[`HostActions renders HostActions with no actions 1`] = `""`;

exports[`HostActions renders HostActions with one action 1`] = `
<a
  className="btn btn-sm btn-default"
  href="/1"
>
  action one
</a>
`;
